### ISSUE TITLE
Keyboard freeze have to regain focus with mouse click

### ISSUE THREAD

**Issue by advice2020** [doublecmd/doublecmd#1408]
### Version

Double Commander 
1.2.0 alpha
gtk
portable
Debian based Linux distro

### Description

All of a sudden recently I have noticed quite a few times that I get these scenarios where keyboard freezes.  I cannot do any navigation in file pane at all, I have to click file pane with mouse to get access to keyboard again.  It is not that DC window itself has lost focus, it seems to be within the program.

Noticed while using a renaming based Lua tool I thought it might have something to do with this but also noticed this when “tab” key toggled between file panes.  The other day it was happening again a couple of times I think when I was cycling through tabs, the cursor was stuck highlighting the top item up arrow.
Have not been able to narrow it down but if anyone else has noticed anything similar, please post info here.
I have not experienced this before, just past week or so.
Sorry I don't have more to help with replicating behavior at the moment.

Thank You

**Comment by JonnyTech** [comment]
Yes, I get that often, I am not at my computer but IIRC, pressing `ALT+F1` to bring up the drive selection list works then I can continue using the keyboard normally. I, too, cannot replicate the issue willingly.

**Comment by ags1234** [comment]
I also experience this. I would describe it like this: even if the keyboard works fine, even if DC window has focus, sometimes, can't reproduce when, it seems randomly, DC does not respond to keyboard commands. I, too, have to click inside the window to make it respond to keyboard. Also, this is not something new for me, it is happening for years.
Could it be related to OLE, where user drags objects outside of DC or inside?

**Comment by wartum** [comment]
Maybe it can occur in different scenarios but this is one of those:


1. Open new tab and set Tab Options -> Locked with Directories Opened in New Tabs
2. Start typing letters to select a directory inside locked directory (I have it configured as Letters -> Quick Search, but Quick Filter also works)
3. Click enter to open this directory in new tab
4. After in opens it's impossible to use keyboard to move around. You can use ctrl+tab to switch tabs, which will "reactivate" keyboard driven navigation

Issue doesn't seem to reproduce if the tab is not locked.
My OS is Win10

Check please if those steps are reproducible on your side.

**Comment by JonnyTech** [comment]
Not the OP, but I do not use locked directories..

**Comment by wartum** [comment]
After seeing your reply, I've experimented a bit trying to reproduce it without having locked tab and it seems, indeed that locking have nothing to do, it's just the fact of creating new tab after using Quick Search / Quick Filter.

So here's second scenario:
1. Select any directory by typing first couple of letters of its' name
2. Use Ctrl + Up Arrow to open this directory in new tab
3. Use Ctrl + Tab to switch to newly created tab
4. Keyboard navigation inside this particualr tab doesn't work

**Comment by ags1234** [comment]
I confirm everything @wartum said. This is a bug.
My details:
```
Double Commander
Version: 1.2.0 alpha
Revision: 11111
Commit: a50a39b4b
Build date: 2023/12/21
Lazarus: 2.2.6.0
FPC: 3.2.2
Platform: x86_64-Win64-win32/win64
OS version: Windows 11 23H2 x86_64

```

**Comment by JonnyTech** [comment]
I cannot confirm on Linux. There is indeed a keyboard focus issue but I still cannot work out the steps to reproduce it.

```
Double Commander
Version: 1.2.0 alpha
Revision: 11097
Commit: bd9285a
Build date: 2023/12/16
Lazarus: 2.2.6.0
FPC: 3.2.2
Platform: x86_64-Linux-qt5
OS version: MX 23.1 Libretto

Widgetset library: Qt 5.15.8, libQt5Pas 5.6.1
```

**Comment by wartum** [comment]
I was able to reproduce that on Fedora 39 with the steps I described previously
```
Double Commander
Version: 1.0.11
Revision: 272
Commit: 0b8ba3838
Build date: 2023/04/02
Lazarus: 2.2.2.0
FPC: 3.2.2
Platform: x86_64-Linux-qt5
OS version: Fedora release 39 (Thirty Nine)

Widgetset library: Qt 5.15.11, libQt5Pas 5.6.1
```

**Comment by MostlyOffline** [comment]
Another way to reproduce this bug:
Switch to a TAB with at least 1 history step.
1. Type a letter or few, one is enough. It doesn't matter whether a match in a directory or a file is found. The issue will occur either way.
2. Press ALT+Left or ALT+Right to go back/forward in TAB's history. 
3. Keyboard is now "locked".
4. Now with F2-Rename, CTRL+F1, Switch to another TAB with CTRL+TAB/Shift+CTRL+TAB, and perhaps a few other actions, the keyboard will "unlock" again. But for example CTRL+F2/3/4 for sorting, doesn't unlock it.

**Comment by JonnyTech** [comment]
These steps reproduce a keyboard lockout for me:

 - press <kbd>ctrl</kbd>+<kbd>q</kbd> to open quick view in opposite panel
 - select a file to view its contents
 - press <kbd>del</kbd> then <kbd>enter</kbd> to delete the file
 - keyboard is now locked out
 
To get keyboard control back:
 - press <kbd>ctrl</kbd>+<kbd>f1</kbd> to force focus back to panel
or
 - press <kbd>alt</kbd>+<kbd>tab</kbd> to switch away then switch back

```
Double Commander
Version: 1.2.0 alpha
Revision: 11261
Commit: 1b39937
Build date: 2024/03/16
Lazarus: 2.2.6.0
FPC: 3.2.2
Platform: x86_64-Linux-qt5
OS version: MX 23.2 Libretto

Widgetset library: Qt 5.15.8, libQt5Pas 5.6.1
```

The GTK version does not exhibit this problem.

**Comment by advice2020** [comment]
Wow, thank you guys for all the feedback, this might be my most active post to date.
Was using QT5/Laz3.0.0.3 compile recently and noticed this happening quite a few times when using the Directory Hotlist + TreeViewMenu.  When DC would navigate to the location in the DHL when it would get back into that location in the file pane it would keep freezing with cursor highlighting the “..” at the top.  Kept having to click inside the window with mouse to get me focus back.

**Comment by advice2020** [comment]
Thank everyone again for all the responses.
Something that I was trying to test generated this issue.
The actions I describe below are not common, but just providing to maybe help narrow down the cause of this in general.

`Steps To Reproduce Issue:`
-Set any hotkey to “cm_Open” for this example say your letter “O”
-Allow this hotkey to work in your Quick Search/Quick Filter field
-If you activate your Quick Filter and type in a folders name
-When that folder name is under the cursor, hit letter “O” to open that folder while focus is still in filter field
-It navigates/opens that folder but then the cursor is frozen at the top
-You have to click in the file pane with mouse to get the focus back

Hopefully this was helpful
Thank You

**Comment by chrapuch** [comment]
A possible woraround:
https://github.com/doublecmd/doublecmd/issues/1717#issuecomment-2337893457

**Comment by cvzakharchenko** [comment]
> A possible workaround: [#1717 (comment)](https://github.com/doublecmd/doublecmd/issues/1717#issuecomment-2337893457)

Doesn't help.

One more way to reproduce:
- Enter a directory with some hierarchy
- Ctrl+B to go into Flat Mode
- Ctrl+S to search for a file
- Search for a random file somewhere deep in hierarchy
- Ctrl+B to exit Flat Mode

After those steps DC ignores arrow keys, tab or backspace. This state can be broken off with hitting Ctrl+S twice.

Version info:

Double Commander
Version: 1.1.17 gamma
Revision: 548
Commit: f4189f213
Build date: 2024/08/19
Lazarus: 3.5.0.0
FPC: 3.2.2
Platform: x86_64-Win64-win32/win64
OS version: Windows 11 23H2 x86_64

**Comment by ags1234** [comment]
I can reproduce what @cvzakharchenko said above in:

Double Commander
Version: 1.2.0 alpha
Revision: 11458
Commit: 9e26803e5
Build date: 2024/09/13
Lazarus: 3.4.0.0
FPC: 3.2.2
Platform: i386-Win32-win32/win64
OS version: Windows 11 23H2 x86_64

**Comment by chfriedrich** [comment]
Exact same problem here.
I first experienced it with v1.1.x installed from a binary. Then I tried building v1.2.0 myself on Windows 11, but this did not change anything.

Double Commander
Version: 1.2.0 alpha
Revision: 11790
Commit: e853cdadd
Build date: 2025/01/17
Lazarus: 3.6.0.0
FPC: 3.2.2
Platform: x86_64-Win64-win32/win64
OS version: Windows 11 24H2 x86_64

**Comment by MostlyOffline** [comment]
Some of these I can't reproduce anymore but https://github.com/doublecmd/doublecmd/issues/1408#issuecomment-1889227905 still triggers it. Wonder if some were fixed by accident because there were no comments from the devs on this.
